File: reverseVowelsOfAString/rvoas.java
class Solution {
    public String reverseVowels(String s) {
        int left = 0;
        int right = s.length()-1;
        String vowels = "eaiouAEIOU";
        char[] li = s.toCharArray();
        while(left<right){
            if(vowels.contains(li[left]+"") && vowels.contains(li[right]+"")){
                char temp = li[left];
                li[left] = li[right];
                li[right] = temp;
                left++;
                right--;
            }
            if(!vowels.contains(li[left]+"")) left++;
            if(!vowels.contains(li[right]+"")) right--;
        }
        return new String(li);
    }
}

File: reverseVowelsOfAString/rvoas.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        left,right = 0,len(s)-1
        li = list(s)
        arr = ['a','e','i','o','u','A','E','I','O','U']
        while left<right:
            if li[left] in arr and li[right] in arr:
                li[left],li[right] = li[right],li[left]
                left+=1
                right-=1
            if not li[left] in arr:left+=1
            if not li[right] in arr:right-=1         
        return ''.join(li)